### ISSUE TITLE
C# WebSocket SendAsync Exception

### ISSUE THREAD

**Issue by GASA94** [dotnet/runtime#56409]
Im executing a net core application using WebSocketClient to communicate with Java Springboot websocketserver.
Im sending over 20000 messages over the websocket from C#:
 
```c#
await client.SendAsync(new ArraySegment<byte>(bytes), WebSocketMessageType.Text, true, _cancellationToken);
```

However, sometimes, certain messages give the next exception:
```
    The Specified argument was out of the range of valid values
```
StackTrace:
```
at System.Net.WebSOckets.ManagedWebSocket.<SendFrameFallbackAsync>d_61.MoveNext()
    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
    at MyWebSocket.ClientHandler.<SendMessage>d_8.MoveNext() in C:\MyProjects\MyWebSocket\ClientHandler.cs:line 71
```

The line corresponds to the Send async placed above.

I have already tried to increase the TextMessage buffer size on the java project, but the exception keeps happening, regardless of byte size. It doesnt happen with all messages, only some of them.
This doesnt stop the java springboot from keep receiving messages and process them, so i believe the issue lies in the way the message is send in the C# client.

Any thoughts will be greatly appreciated.

**Comment by CarnaViire** [comment]
Could you please provide a minimal repro @GASA94 ?

**Comment by GASA94** [comment]
> Could you please provide a minimal repro @GASA94 ?

The code used to send messages:

`public async SendMessage(MyMessage message){
    string messagedata = JsonConvert.SerializeObject(message);
    SendCommand command = new SendCommand("message", messagedata);
    string msg = JsonConvert.SerializeObject(command);
    byte[] bytes = Encoding.UTF8.GetBytes(msg);
    await client.SendAsync(new ArraySegment<byte>(bytes), WebSocketMessageType.Text, true, _cancellationToken);
}`

Send Command stores a simple string and the command to send, it is then converted into a json to send it to the java application, using the websocket,
MyMessage object simply has a string and a long value, they always have a value.

To send several messages a visitor is used to scan a folder and it just saves the file names in the MyMessage object and a long value that serves as a counter (each file has a unique long value). Hope this helps, apologies if im doing a mistake here, this is my first time using this platform.

**Comment by karelz** [comment]
Triage: @GASA94 do you have a (minimal) repro we can just run? Even against public server if needed.

**Comment by stephentoub** [comment]
What version are you running on?  This sounds like https://github.com/dotnet/runtime/issues/39198, which was fixed in .NET 5 in https://github.com/dotnet/runtime/pull/39199, and which was only just backported to 3.1 in https://github.com/dotnet/corefx/pull/42950 (so it hasn't been released in a 3.1 patch yet).

**Comment by GASA94** [comment]
> What version are you running on? This sounds like #39198, which was fixed in .NET 5 in #39199, and which was only just backported to 3.1 in [dotnet/corefx#42950](https://github.com/dotnet/corefx/pull/42950) (so it hasn't been released in a 3.1 patch yet).

that does sound very similar to my problem, the proyect is on netcore 3.1, so i will have to wait for the patch then.
